api/models/group.py

```diff
@@ -1,31 +1,45 @@
 from django.db import models
 
 from api.models.days_time_slots import DayAndTimeSlot
-from api.models.languages_levels import CommunicationLanguageMode, TeachingLanguageAndLevel
+from api.models.languages_levels import (
+    CommunicationLanguageMode,
+    TeachingLanguage,
+    TeachingLanguageAndLevel,
+)
 from api.models.people import Coordinator, Student, Teacher
 from api.models.statuses import GroupStatus
 
 
+class GroupCommon(models.Model):
+    """Abstract model for attributes shared by regular groups and speaking clubs."""
+
+    coordinators = models.ManyToManyField(Coordinator)
+    students = models.ManyToManyField(Student)
+    teachers = models.ManyToManyField(Teacher)
+    # group chat created manually by the coordinator/teacher
+    telegram_chat_url = models.URLField(blank=True, null=True)
+
+    class Meta:
+        abstract = True
+
+
 # TODO think about group clusters for auto-constructed groups
-class Group(models.Model):
+class Group(GroupCommon):
+    """Model for a regular language group."""
+
     availability_slot = models.ManyToManyField(DayAndTimeSlot)
     communication_language_mode = models.ForeignKey(
         CommunicationLanguageMode, on_delete=models.PROTECT
     )
     is_for_staff_only = models.BooleanField(default=False)
-    language_and_level = models.ForeignKey(TeachingLanguageAndLevel, on_delete=models.CASCADE)
-    lesson_duration = models.IntegerField()
+    language_and_level = models.ForeignKey(TeachingLanguageAndLevel, on_delete=models.PROTECT)
+    lesson_duration = models.PositiveSmallIntegerField()
     status = models.ForeignKey(GroupStatus, on_delete=models.PROTECT)
```

Comment:
That's an interesting one, because language levels won't change, but languages can. And if we add a new language, the program logic won't be affected. So languages should definitely stay in the DB, while levels can be an enum. And each group has a language and a level as its core characteristic. I have to do some research of the docs and see if that is doable.

We could ofc put "language" in one column and "level" in another.

Reply:
OK I guess we'll leave the languages in DB for now, and levels as choices within `LanguageAndLevel`